runtime/src/accounts.rs

```diff
@@ -1702,6 +1725,41 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_accounts_remove() {
+        let paths = get_tmp_accounts_path!();
+        let accounts = AccountsDB::new(0, &paths.paths);
+        let mut pubkeys0: Vec<Pubkey> = vec![];
+        create_account(&accounts, &mut pubkeys0, 0, 100, 0);
+        update_accounts(&accounts, &pubkeys0, 0, 99);
+        {
+            let stores = accounts.storage.read().unwrap();
+            assert_eq!(stores.len(), 1);
+            assert_eq!(stores[0].count.load(Ordering::Relaxed), 100);
+        }
+        accounts.add_fork(1, Some(0));
+        let mut pubkeys1: Vec<Pubkey> = vec![];
+        create_account(&accounts, &mut pubkeys1, 1, 100, 0);
+        update_accounts(&accounts, &pubkeys1, 1, 99);
+        {
+            let stores = accounts.storage.read().unwrap();
+            assert_eq!(stores.len(), 1);
+            assert_eq!(stores[0].count.load(Ordering::Relaxed), 200);
+        }
+        accounts.remove_accounts(0);
+        {
+            let stores = accounts.storage.read().unwrap();
+            assert_eq!(stores.len(), 1);
+            assert_eq!(stores[0].count.load(Ordering::Relaxed), 100);
+        }
```

Comment:
Can we test adding a fork after removing and also loading some of the accounts that we have deleted and not-deleted.

Reply:
Sure..will update the test